refactor_show_prototype.jl

```diff
@@ -0,0 +1,136 @@
+using Pkg
+Pkg.activate(@__DIR__)
+using Latexify
+import Base: show
+using SnoopCompile
+
+
+const Lmime = MIME"text/latexify"
+const lmime = MIME("text/latexify")
+
+const opdict = (;
+    :* => :Multiplication,
+    :/ => :Division,
+    :+ => :Addition,
+    :- => :Subtraction,
+    :^ => :Pow,
+)
+for T in values(opdict)
+    println(T)
+end
+abstract type AbstractLatexifyOperation end
+for T in values(opdict)
+    @eval(
+        struct $(T){S} <: AbstractLatexifyOperation
+            args::S
+        end
+    )
+end
```

Comment:
An alternative is 
```julia
struct Operation{S, T} where {S<:Val, T}
args::T
end
```
and the likes of
```julia
function show(io::IO, mime::Lmime, Operation{Val{:+}, T}) where T
```

Reply:
Good idea. I modified it a bit though. I added the ability to overload on `head` and I removed type specification on the args since I don't want the compiler to over-specialise on that (faster runtime at the expense of compiletime).